src/domain/books/BooksPage.tsx

```diff
@@ -11,20 +11,23 @@ export const BooksPage = (): JSX.Element => {
 
   useEffect(() => {
     dispatch(getBooks());
-  }, [Bookdata]);
+  }, []);
 
-  return (
-    <div className="book-page-container">
-      <>{console.log(Bookdata)}</>
-      {Bookdata.length === 0 ? (
-        <div>Loading...</div>
-      ) : (
+  const renderBooks = () => {
+    if (Bookdata.loading === "loaded") {
+      return (
         <div className="book-list">
-          {Bookdata.map((book) => (
+          {Bookdata.books.map((book) => (
             <BookCard {...book} key={book.id} />
           ))}
         </div>
-      )}
-    </div>
-  );
+      );
+    } else if (Bookdata.loading === "loading") {
+      return <div>{Bookdata.loading}...</div>;
+    } else {
+      return <div>An error occured...</div>;
+    }
+  };
```

Comment:
This is OK.
But a general rule of thumb is to always have the "success" case at the end. In this case, booksList should be returned at last and before that all the guard clauses are placed.

Another thing, just use If and return. No need for else-if and else blocks. They are redundant.